### PR TITLE
Fixes bug 2428 

### PR DESCRIPTION
**Pull request by mes5k**
To fix this bug I add an additional check to checkInvalidWaitingState that checks if the pod phase is Failed and throw an exception if it is. A test validating this behavior has been included and I have confirmed that the test fails without this fix.

### CODE REVIEW

File: modules/nextflow/src/test/groovy/nextflow/k8s/client/K8sClientTest.groovy [hunk 1]
```diff
@@ -845,4 +845,121 @@ class K8sClientTest extends Specification {
...
+                            "type": "ContainersReady"
+                        },
+                        {
+                            "lastProbeTime": null,
+                            "lastTransitionTime": "2021-11-03T21:37:45Z",
+                            "status": "True",
+                            "type": "PodScheduled"
+                        }
+                    ],
+                    "containerStatuses": [
```
Comment (pditommaso) [inline modules/nextflow/src/test/groovy/nextflow/k8s/client/K8sClientTest.groovy:905]: Maybe the `conditions` can be removed to simplify the test sample

**Review by pditommaso — COMMENTED** [review]
Thanks for reporting this patch! I've left a comment

File: modules/nextflow/src/test/groovy/nextflow/k8s/client/K8sClientTest.groovy (hunk 1 above)
Comment (mes5k) [reply modules/nextflow/src/test/groovy/nextflow/k8s/client/K8sClientTest.groovy:873]: Sure, just removed them.

Comment (pditommaso) [reply modules/nextflow/src/test/groovy/nextflow/k8s/client/K8sClientTest.groovy:873]: Awesome. Thanks Mike!

**Pull request merged**